CRM/Contribute/BAO/Contribution.php

```diff
@@ -2561,6 +2561,11 @@ public function composeMessageArray(&$input, &$ids, &$values, $returnMessageText
             if ($membership_status == 'Pending' && $membership->is_pay_later == 1) {
               $values['is_pay_later'] = 1;
             }
+            // Pass amount to floatval as string '0.00' is considered a
+            // valid amount and includes Fee section in the mail.
+            if (isset($values['amount'])) {
+              $values['amount'] = floatval($values['amount']);
+            }
```

Comment:
**api_v3_ContributionPageTest**: 
`testSubmitMembershipBlockIsSeparatePaymentZeroDollarsPayLaterWithEmail` and 
`testSubmitMembershipBlockNotSeparatePaymentZeroDollarsWithEmail` asserts this change after modification to the second file included in this PR.